Makefile

```diff
@@ -114,7 +114,7 @@ include $(TMK_PATH)/protocol/lufa.mk
 include $(TMK_PATH)/common.mk
 include $(TMK_PATH)/rules.mk
 
-GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags 2>/dev/null || date +"%Y-%m-%d")
-
+GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags 2>/dev/null)
+BUILD_DATE := $(shell date +"%Y-%m-%d-%H:%M:%S")
```

Comment:
There's one issue I see with this: if the firmware is not built from a git checkout, `QMK_VERSION` will be empty. I'd suggest keeping the new `BUILD_DATE`, but preserve the former behaviour of `QMK_VERSION` too, by having an `|| echo ${BUILD_DATE}` at the end of `GIT_VERSION`.

This way we get the best of both.